lib/ja_serializer/dsl.ex

```diff
@@ -217,15 +230,17 @@ defmodule JaSerializer.DSL do
   using super, or without the DSL just returning a map.
 
   """
-  defmacro attributes(atts) do
+  defmacro attributes(atts) when is_list(atts) do
     quote bind_quoted: [atts: atts] do
       # Save attributes
       @attributes @attributes ++ atts
 
       # Define default attribute function, make overridable
       for att <- atts do
+        @compile {:inline, [{att, 1}, {att, 2}]}
+
         def unquote(att)(m),    do: Map.get(m, unquote(att))
-        def unquote(att)(m, c), do: apply(__MODULE__, unquote(att), [m])
+        def unquote(att)(m, c), do: unquote(att)(m)
```

Comment:
`apply` cannot be inlined compile time. By calling the function directly it allows compiler to inline the function.